specs/CustomStoragePartition.md

```diff
@@ -0,0 +1,208 @@
+        void CoreWebView_Created()
+        {
+           // ...
+           // other WebView setup like add event handlers, update settings.
+           
+           // Sets custom storage partition identified by the partitionId, which uniquely
+           // identifies an application context.
+           string partitionId = "Partition 1";
+           m_webview.CustomStoragePartitionId = partitionId;
+           
+           // Navigate to start page
+           m_webview.Navigation(startPage);
+        }
+        
+        // Clears all data in custom storage partition identified by the partitionId.
+        // Called when the application context is removed.
+        async void ClearPartitionData(string partitionId)
+        {
+          await m_webview.Profile.ClearCustomStoragePartitionDataAsync(partitionId);
+          MessageBox.Show(this, "Completed", "Clear Custom Partition Data");
+        }
+
+```
+
+# API Details
+## Win32 C++
+```
+interface ICoreWebView2_18 : ICoreWebView2_17 {
+  /// Gets the `CustomStoragePartitionId` property.
+  [propget] HRESULT CustomStoragePartitionId([out, retval] LPWSTR* customStoragePartitionId);
+
+  /// Sets the `CustomStoragePartitionId` property.
+  /// This feature requires enabling 2 experimental browser features to work properly.
+  /// These features will be enabled by default in the future.
+  /// Before these features are enabled by default, please enable them by ensuring
+  /// `--enable-features=ThirdPartyStoragePartitioning,PartitionedCookies` is set in
+  /// `AdditionalBrowserArguments` in `CoreWebView2EnvironmentOptions` used to create
+  /// CoreWebView2Environment. If these features are not enabled, all data are treated
+  /// as unpartitioned and stored in the global default location for the profile.
+  /// When it is set, the page in the WebView will act as if the page were hosted in a
+  /// top level site uniquely associated with the `partitionId` and have a separate 
+  /// storage partition as described in https://developer.chrome.com/docs/privacy-sandbox/storage-partitioning/
+  /// and separete cookie partition as described in https://developer.chrome.com/docs/privacy-sandbox/chips/
+  /// with all cookies partitioned.
+  /// If `customStoragePartitionId` is nullptr or empty string, the
+  /// `CustomStoragePartitionId` will be reset and the page inside the WebView
+  /// will work normally with data treated as unpartitioned.
+  /// The `customStoragePartitionId` parameter is case sensitive. The default is
+  /// an empty string. There is no restriction on the length or what characters
+  /// can be used in partition id.
+  /// The change of the custom storage partition id will be applied to new 
+  /// page or iframe navigations and not impact existing pages and iframes.
+  /// To avoid accidentally use new partition id for pending navigations of old page
+  /// or iframe, it is recommended to create a new WebView for new partition instead
```

Comment:
This sentence suggests that the value is captured when the navigation ends, but previous sentence suggests that it is captured when the navigation *starts*.

Reply:
The value is captured at navigation start, we should change the documentation to: To avoid accidentally use new partition id for new page or iframe navigations started from the old page, ...

Reply:
Please update to make clear its navigation start. And explicitly call out frames.